src/Searchbar.vue

```diff
@@ -35,43 +27,42 @@ export default {
         const headingEntries = [];
 
         Object.entries(headings).forEach(([id, text]) => {
-            const info = [title].concat(keywords).concat(text).join(' ');
-            if (allMatch(info, regexes)) {
-              if(someMatch(text, regexes)) {
-                hasMatchingHeading = true;
-              }
-              headingEntries.push({
-                heading: { id, text },
-                keywords,
-                src,
-                title,
-              });
-            }
-          });
-          if (!hasMatchingHeading) {
-            const info = [title].concat(keywords).join(' ');
-            if(allMatch(info, regexes)) {
-              matches.push(entry);
+          const info = [title].concat(keywords).concat(text).join(' ');
+          if (regexes.every(regex => regex.test(info))) {
+            if (regexes.some(regex => regex.test(text))) {
+              hasMatchingHeading = true;
             }
+            headingEntries.push({
+              heading: { id, text },
+              keywords,
+              src,
+              title,
+            });
+          }
+        });
+        if (!hasMatchingHeading) {
+          const info = [title].concat(keywords).join(' ');
+          if (regexes.some(regex => regex.test(info))) {
+            matches.push(entry);
```

Comment:
Will this logic as well? I got rid of the `hasMatchingHeading` boolean by rearranging the logic.

```javascript
this.data.forEach((entry) => {
        const { headings, src, title } = entry;
        const keywords = entry.keywords || '';
        const headingEntries = [];

        const info = [title].concat(keywords).join(' ');
        if (regexes.every(regex => regex.test(info))) {
          matches.push(entry);
        }
        
        Object.entries(headings).forEach(([id, text]) => {
          const info = [title].concat(keywords).concat(text).join(' ');
          if (regexes.every(regex => regex.test(info))) {
            headingEntries.push({
              heading: { id, text },
              keywords,
              src,
              title,
            });
          }
        });
        
        matches = matches.concat(headingEntries);
      });
```